packages/@cdktf/provider-generator/lib/get/constructs-maker.ts

```diff
@@ -294,79 +366,24 @@ export class ConstructsMaker {
     }
 
     if (!this.isJavascriptTarget || this.options.outputJsii) {
-      for (const target of this.targets) {
-        // these are the module dependencies we compile against
-        const deps = ["@types/node", "constructs", "cdktf"];
-        const opts: srcmak.Options = {
-          entrypoint: target.fileName,
-          deps: deps.map((dep) =>
-            path.dirname(require.resolve(`${dep}/package.json`))
-          ),
-          moduleKey: target.moduleKey,
-        };
-
-        // used for testing.
-        if (this.options.outputJsii) {
-          opts.jsii = { path: this.options.outputJsii };
-        }
-
-        if (this.isPythonTarget) {
-          opts.python = {
-            outdir: this.codeMakerOutdir,
-            moduleName: target.srcMakName,
-          };
-        }
-
-        if (this.isJavaTarget) {
-          opts.java = {
-            outdir: ".", // generated java files aren't packaged, so just include directly in app
-            package: `imports.${target.srcMakName}`,
-          };
-        }
-
-        if (this.isCsharpTarget) {
-          opts.csharp = {
-            outdir: this.codeMakerOutdir,
-            namespace: target.srcMakName,
-          };
-        }
-
-        if (this.isGoTarget) {
-          // TODO: check if needed for modules somehow
-          // const targetType = target.isProvider ? 'provider' : 'module';
-
-          // jsii-srcmac will produce a folder inside this dir named after "packageName"
-          // so this results in e.g. .gen/hashicorp/random
-          const outdir = path.join(
-            this.codeMakerOutdir,
-            target.namespace ?? ""
+      const numberOfWorkers =
+        this.options.jsiiParallelism === -1
```

Comment:
Maybe change this (or instead use `Math.max()` for the value below in line 372) to ensure passing `-2` or `0` does not cause an endless loop or crash?
```suggestion
        this.options.jsiiParallelism <= 0 || typeof this.options.jsiiParallelism === "undefined"
```

Reply:
Oh, just noticed: passing `undefined` would create an array with the length `undefined`, wouldn't it? (before my suggested change)

Reply:
Nice! Looks good